pulpcore/cli/deb/content.py

```diff
@@ -0,0 +1,321 @@
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--essential",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--installed-size",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--maintainer",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--origin",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--package",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageReleaseComponentContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--source",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--tag",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--version",
+        allowed_with_contexts=(
+            PulpDebInstallerPackageContext,
+            PulpDebPackageContext,
+        ),
+    ),
+    pulp_option(
+        "--component",
+        allowed_with_contexts=(
+            PulpDebInstallerFileIndexContext,
+            PulpDebPackageIndexContext,
+            PulpDebReleaseComponentContext,
+        ),
+    ),
+    pulp_option(
+        "--release-component", allowed_with_contexts=(PulpDebPackageReleaseComponentContext,)
+    ),
+]
+lookup_options = [
+    href_option,
+    pulp_option(
+        "--relative-path",
+        callback=_relative_path_callback,
+        expose_value=False,
+        allowed_with_contexts=(
+            PulpDebGenericContentContext,
+            PulpDebInstallerFileIndexContext,
+            PulpDebPackageIndexContext,
+            PulpDebPackageContext,
+            PulpDebReleaseFileContext,
+        ),
+    ),
+    pulp_option(
+        "--sha256",
+        callback=_sha256_callback,
+        expose_value=False,
+        allowed_with_contexts=(
+            PulpDebGenericContentContext,
+            PulpDebInstallerFileIndexContext,
+            PulpDebPackageIndexContext,
+            PulpDebPackageContext,
+            PulpDebReleaseFileContext,
+        ),
+    ),
+]
+
+content.add_command(list_command(decorators=list_options))
+content.add_command(show_command(decorators=lookup_options))
+# create assumes "there exists an Artifact..."
+# create is defined for package and expects the sha256 of the artifact.
+create_options = [
+    pulp_option(
+        "--relative-path",
```

Comment:
We probably don't want the `--relative-path` option for the create and upload commands, since unlike with pulp_rpm, pulp_deb is designed to auto-generate the relative path (at least for packages).